### PR TITLE
Make the game look and play better

### PR DESCRIPTION
**Pull request by sharkwouter**
This PR:
- Replaces all assets. It now uses shells and has a background.
- Make it visible how the grid changes when moving a block.
- Add score and move display
- Cleans up the code in general
- Use shells instead of gems. The names have not been replaced everywhere yet. There are some more name changes in the next PR.

Unfortunately, it also includes the PSP changes. This doesn't work at the moment. I need to fix the PSP's SDL implementation to make it work as well. Annoying. Should I remove that part or just leave it be?

### CODE REVIEW

**Comment by sharkwouter** [comment]
This also breaks Vita support. We could merge this and create an issue for fixing/deleting PSP and Vita support or fix this before merging.

File: src/Board.cpp [hunk 1]
```diff
@@ -108,8 +108,13 @@ bool Board::swap(SDL_Point p1, SDL_Point p2) {
     }
 }
 
-std::vector<std::vector<Gem>> Board::getGemsAfterSwap(std::vector<std::vector<Gem>> gems, SDL_Point p1, SDL_Point p2) {
-    Gem type1 = gems[p1.x][p1.y];
+std::vector<std::vector<Shell>> Board::getGemsAfterSwap(std::vector<std::vector<Shell>> gems, SDL_Point p1, SDL_Point p2) {
```
Comment (protheory8) [inline src/Board.cpp:111]: Should be named `getShellsAfterSwap` now?

File: src/Board.cpp [hunk 2]
```diff
@@ -140,11 +145,11 @@ bool Board::isWithinBounds(SDL_Point point) {
     return result;
 }
 
-std::vector<std::vector<Gem>> Board::getGemsCopy() {
-    std::vector<std::vector<Gem>> copy;
+std::vector<std::vector<Shell>> Board::getGemsCopy() {
```
Comment (protheory8) [inline src/Board.cpp:148]: `getShellsCopy`?

File: src/main.cpp [hunk 3]
```diff
@@ -10,6 +10,12 @@
 #include "states/GameState.hpp"
 
 
+#ifdef __PSP__
+    #include "pspkernel.h"
+    PSP_MODULE_INFO("SDL App", 0, 1, 1);
```
Comment (protheory8) [inline src/main.cpp]: Is this supposed to be called `SDL App`?

File: src/states/GameState.hpp [hunk 4]
```diff
@@ -9,6 +9,9 @@
 class GameState : public BaseState {
 private:
     BoardManager board;
+    TextureManager textures;
+
+    std::string background_image = "assets/images/background.jpg";
```
Comment (protheory8) [inline src/states/GameState.hpp:14]: Probably could be a constant in `constants.hpp` instead, but it's fine to leave as it is.

**Comment by protheory8** [comment]
> Should I remove that part or just leave it be?

Leave it be I guess.

File: src/states/GameState.hpp (hunk 4 above)
Comment (sharkwouter) [reply src/states/GameState.hpp:14]: Maybe, I want to move it to a configuration file in the future.

**Pull request merged**